ui-components/urldialog.md

```diff
@@ -7,91 +7,90 @@ keywords: dialog urldialog api
-| onAction | (api) => void | optional | Function invoked when a custom button is clicked. |
-| onCancel | (api) => void | optional | Function invoked when the dialog is cancelled. |
-| onClose | () => void | optional | Function invoked when the dialog has been closed. |
-| onMessage | (api, data) => void | optional | Function invoked when a message is received from the external page. |
-
-### Footer components
+| buttons | FooterButton[] | optional | An optional array of [footer buttons](#footerbuttons) to render in the footer of the dialog. |
+| onAction | `(dialogApi, details) => void` | optional | Function invoked when a **Custom** type footer button is clicked. |
+| onCancel | `(dialogApi) => void` | optional | Function invoked when the dialog is cancelled. The dialog header's close button and a **Cancel** type footer button invoke this function. |
+| onClose | `() => void` | optional | Function invoked when the dialog is closed. The dialog header's close button, a **Cancel** type footer button and the dialog instance API's `close()` method invoke this function. |
+| onMessage | `(dialogApi, details) => void` | optional | Function invoked when a message is received from the external page. |
 
-#### Button
+For more information on the `dialogApi` object that is passed to some of the configuration options, see the [URL dialog instance API](#urldialoginstanceapi) documentation.
 
-The following configuration is used to create a button inside the dialog footer:
+### Footer buttons
 
-```js
-var buttonConfig = {
-  type: 'button',
-  name: string,
-  text: string,
-  icon: string,
-  disabled: boolean,
-  primary: boolean,
-  align: 'start' | 'end'
-}
-```
+A **button** is a clickable component that can contain text or an icon. There are two types of buttons (primary and secondary buttons), though the only difference is that they are styled differently. Primary buttons are intended to stand out. The color will depend on the chosen [skin]({{site.baseurl}}/general-configuration-guide/customize-ui/#skins).
 
-**Name:** The name property on the button is used as an ID attribute to identify the dialog component. For example, when `name: foobutton` is defined and a user clicks on that button, the dialog `onAction()` handler will fire and provide an object containing the name of the dialog component, e.g., `details.name = 'foobutton'`. This will allow developers to create a click handler for **foobutton**.
+#### Configuration
 
-**Text:** This will be the text displayed on the button. For example, `text: ‘do magic’` will create a button with text **do magic**.
+| Name | Type | Requirement | Description |
+| ---- | ---- | ----------- | ----------- |
+| type | `'button'` | required | The component type. Must be `'button'`. |
+| text | string | required | Text to display in the button **if `icon` is not specified**. Also used for the button's `title` attribute. |
```

Comment:
The bold markup around `if icon is not specified` hides that icon is a keyword and I'm not sure it makes sense to have that much bold? Maybe just make `if` bold or remove it altogether?